lib/ansible/modules/network/cloudengine/ce_snmp_user.py

```diff
@@ -0,0 +1,1074 @@
+      </snmp>
+    </filter>
+"""
+# merge snmp v3 aaa local user
+CE_MERGE_SNMP_V3_LOCAL_USER = """
+    <config>
+      <snmp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+        <localUsers>
+          <localUser operation="merge">
+            <userName>%s</userName>
+            <authProtocol>%s</authProtocol>
+            <authKey>%s</authKey>
+            <privProtocol>%s</privProtocol>
+            <privKey>%s</privKey>
+          </localUser>
+        </localUsers>
+      </snmp>
+    </config>
+"""
+# create snmp v3 aaa local user
+CE_CREATE_SNMP_V3_LOCAL_USER = """
+    <config>
+      <snmp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+        <localUsers>
+          <localUser operation="create">
+            <userName>%s</userName>
+            <authProtocol>%s</authProtocol>
+            <authKey>%s</authKey>
+            <privProtocol>%s</privProtocol>
+            <privKey>%s</privKey>
+          </localUser>
+        </localUsers>
+      </snmp>
+    </config>
+"""
+# delete snmp v3 aaa local user
+CE_DELETE_SNMP_V3_LOCAL_USER = """
+    <config>
+      <snmp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+        <localUsers>
+          <localUser operation="delete">
+            <userName>%s</userName>
+            <authProtocol>%s</authProtocol>
+            <authKey>%s</authKey>
+            <privProtocol>%s</privProtocol>
+            <privKey>%s</privKey>
+          </localUser>
+        </localUsers>
+      </snmp>
+    </config>
+"""
+
+
+class SnmpUser(object):
+    """ Manages SNMP user configuration """
+
+    def netconf_get_config(self, **kwargs):
+        """ Get configure by netconf """
+
+        module = kwargs["module"]
+        conf_str = kwargs["conf_str"]
+
+        xml_str = get_nc_config(module, conf_str)
+
+        return xml_str
+
+    def netconf_set_config(self, **kwargs):
+        """ Set configure by netconf """
+
+        module = kwargs["module"]
+        conf_str = kwargs["conf_str"]
+
+        xml_str = set_nc_config(module, conf_str)
+
+        return xml_str
+
+    def check_snmp_v3_usm_user_args(self, **kwargs):
+        """ Check snmp v3 usm user invalid args """
+
+        module = kwargs["module"]
+        result = dict()
+        result["usm_user_info"] = []
+        need_cfg = False
+        state = module.params['state']
+        usm_user_name = module.params['usm_user_name']
+        remote_engine_id = module.params['remote_engine_id']
+
+        acl_number = module.params['acl_number']
+        user_group = module.params['user_group']
+        auth_protocol = module.params['auth_protocol']
+        auth_key = module.params['auth_key']
+        priv_protocol = module.params['priv_protocol']
+        priv_key = module.params['priv_key']
+
+        local_user_name = module.params['aaa_local_user']
+
+        if usm_user_name:
+
+            if local_user_name:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
```